main/integration-tests/__test__/api-tests/studies/study.test.js

```diff
@@ -0,0 +1,132 @@
+      const studyA = await testFixture.createMyStudy(nonAdminUser.axiosClient, testProjectId);
+
+      // UserB tries to access nonAdminUser's personal study
+      const userB = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const updateRequest = { id: studyA.id, description: 'Sample desc change' };
+      const params = updateStudyParams(studyA.id, updateRequest);
+
+      // EXECUTE & CHECK
+      await expect(userB.axiosClient.get(params.api)).rejects.toMatchObject({
+        response: { status: 404 },
+      });
+    });
+  });
+
+  describe('Get Specific Study API: /api/studies/:id', () => {
+    it("should fail while trying to get another user's personal studies", async () => {
+      // BUILD
+      const nonAdminUser = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const studyA = await testFixture.createMyStudy(nonAdminUser.axiosClient, testProjectId);
+
+      // UserB tries to access nonAdminUser's personal study
+      const userB = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const params = getStudyParams(studyA.id);
+
+      // EXECUTE & CHECK
+      await expect(userB.axiosClient.get(params.api)).rejects.toMatchObject({
+        response: { status: 404 },
+      });
+    });
+
+    it('should pass while trying to get own studies', async () => {
+      // BUILD
+      const nonAdminUser = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+      const studyA = await testFixture.createMyStudy(nonAdminUser.axiosClient, testProjectId);
+      const params = getStudyParams(studyA.id);
+
+      // EXECUTE & CHECK
+      const response = await nonAdminUser.axiosClient.get(params.api);
+      expect(response.data).toHaveProperty('projectId');
+      expect(response.data).toHaveProperty('rev');
+      expect(response.data).toHaveProperty('resources');
+      expect(response.data).toHaveProperty('updatedAt');
+      expect(response.data).toHaveProperty('category');
+      expect(response.data).toHaveProperty('createdAt');
+      expect(response.data).toHaveProperty('updatedBy');
+      expect(response.data).toHaveProperty('description');
+      expect(response.data).toHaveProperty('id');
+      expect(response.data).toHaveProperty('createdBy');
+      expect(response.data).toHaveProperty('name');
+      expect(response.data).toHaveProperty('uploadLocationEnabled');
+      expect(response.status).toEqual(200);
+    });
+  });
+
+  describe('Get Study List API: /api/studies/', () => {
+    it('should pass while trying to get studies list', async () => {
+      // BUILD
+      const newUser = await testFixture.createNonAdminUser(testAdmin.axiosClient, testProjectId);
+
+      // EXECUTE & CHECK
+      const response = await listStudies(newUser.axiosClient, 'Open Data');
+      expect(response).not.toBeUndefined();
```

Comment:
We should check for something more specific. At the very least check that it responds with status 200.

Is the idea here to only test authz? are we adding later another test that actually tests that this api is listing studies? It'd be nice to have assertions that check that some known 'Open data' studies appear in the response.

Reply:
This is still a work in progress. I'll remove the tests that are not pertaining to this bug fix for now.